repl/convert_ast.py

```diff
@@ -7,19 +7,64 @@
 import qwerty_ast
 #################### COMMON CODE FOR BOTH @QPU AND @CLASSICAL DSLs ####################
 
+
 def convert_to_qwerty(py_ast):
     print(ast.dump(py_ast, indent=4))
-    value = unpack_ast(py_ast)
-    if value is None:
-        return "Invalid Expression: Not a qubit literal or Tensor"
-    tensor = qwerty_ast.NodeBox.new_qubit_tensor(py_ast.body[0].value.value)
-    return tensor.get_tensor()
-
-def unpack_ast(ast):
+    if not isinstance(py_ast, (ast.Interactive, ast.Module, ast.Expression)):
+        return "Invalid AST"
+    
+    value = py_ast.body[0]
+    if not isinstance(value, ast.Expr):
+        return "Invalid syntax; expected an expression"
+    
     try:
-        return ast.body[0].value.value
-    except AttributeError:
-        return None
+        return convert_expr(value.value)
+    except Exception as e:
+        return f"Error: {e}"
+    
+#   1) check if the instance is a binOp, if so, check its left or right values and then perform the binOp. only sum is implemented for now
+#   2) check if the instance is a constant and return it as a tensor, return value in a tensor
+
+    
+def convert_expr(node):
+    if isinstance(node, ast.Constant):
+        val = node.value
+        if isinstance(val, str) and all(num in "01" for num in val):
+            tensor = qwerty_ast.NodeBox.new_qubit_tensor(node.value)
+            # print("tensor type:", type(tensor))
+            # print("tensor contents:", dir(tensor))            
+            return tensor
+        else:
+            raise ValueError("Invalid character, e.g. not 1s and 0s")
```

Comment:
This is a very reasonable error message, but you could consider printing out the invalid character itself to make it slightly more helpful